state/runtime/evm/instructions_test.go

```diff
@@ -10,107 +10,376 @@ import (
 	"github.com/0xPolygon/polygon-edge/state/runtime"
 	"github.com/0xPolygon/polygon-edge/types"
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
 	"github.com/stretchr/testify/require"
 )
 
 var (
-	two = big.NewInt(2)
+	two   = big.NewInt(2)
+	three = big.NewInt(3)
+	four  = big.NewInt(4)
+	five  = big.NewInt(5)
 
 	allEnabledForks = chain.AllForksEnabled.At(0)
 )
 
-type twoOperandsArithmetic []struct {
-	a              *big.Int
-	b              *big.Int
+type OperandsLogical struct {
+	operands       []*big.Int
+	expectedResult bool
+}
+
+func testLogicalOperation(t *testing.T, f instruction, test OperandsLogical, s *state) {
+	t.Helper()
+
+	for _, operand := range test.operands {
+		s.push(operand)
+	}
+
+	f(s)
+
+	if test.expectedResult {
+		assert.Equal(t, uint64(1), s.pop().Uint64())
+	} else {
+		assert.Equal(t, uint64(0), s.pop().Uint64())
+	}
+}
+
+type OperandsArithmetic struct {
+	operands       []*big.Int
 	expectedResult *big.Int
 }
 
-func testArithmeticOperation(t *testing.T, f instruction, tests twoOperandsArithmetic) {
+func testArithmeticOperation(t *testing.T, f instruction, test OperandsArithmetic, s *state) {
 	t.Helper()
 
+	for _, operand := range test.operands {
+		s.push(operand)
+	}
+
+	f(s)
+
+	assert.EqualValues(t, test.expectedResult.Uint64(), s.pop().Uint64())
+}
+
+func createState(forks *chain.ForksInTime) (*state, func()) {
+	s, closeFn := getState()
+
+	s.gas = 1000
+	s.config = forks
+	s.host = &mockHost{}
```

Comment:
You have different setup scenarios for `mockHost`, so either provide a delegate parameter that would build specific `mockHost` (and mock required functions) or set it from the outside of this function (as it was previously).